orchagent/pfcactionhandler.cpp

```diff
@@ -271,6 +329,69 @@ PfcWdLossyHandler::~PfcWdLossyHandler(void)
     }
 }
 
+bool PfcWdLossyHandler::getHwCounters(PfcWdHwStats& counters)
+{
+    SWSS_LOG_ENTER();
+
+    static const vector<sai_queue_stat_t> queueStatIds =
+    {
+        SAI_QUEUE_STAT_PACKETS,
+        SAI_QUEUE_STAT_DROPPED_PACKETS,
+    };
+
+    static const vector<sai_ingress_priority_group_stat_t> pgStatIds =
```

Comment:
Can we separate the counters for drop/forward action, and only query the needed counters depends on the pfcwd action?

Reply:
I cannot guarantee that all handlers have a correct implementation of drop/forward handlers, and not getting all counters can hide bugs with traffic not being dropped/forwarded. Besides, it's not polling but a one time action. What's the benefit of not reading 2 counters?

Reply:
It is possible that some vendors might only support one action type eg. drop action, and don't support the counters needed for forward action, though I can also see your point.

Reply:
If some vendor does not support any of counters, then it should be set to 0. It's common practice to set counters to zero. It's easier to extend counters if needed in the future just by extending struct definition instead of declaring new functions.

Reply:
We could set counters to zero, though I still don't think we need to query all the hardware counters for both drop/forward actions regardless of the action type.